src/parser/parsers/generics.py

```diff
@@ -0,0 +1,132 @@
+import utils.json_utils as json_utils
+import os
+
+
+class GenericParser:
+    """
+    Lightly parse the generic data to have more readable keys.
+
+    Validate structures used by the frontend to identify
+    possible failure points in the case they are modified
+    by Valve.
+    """
+
+    def __init__(self, output_dir, generic_data):
+        self.OUTPUT_DIR = output_dir
+        self.generic_data = generic_data
+
+    def _read(self):
+        if not os.path.exists(self.OUTPUT_DIR):
+            return None
+
+        # Read existing generic data from file
+        return json_utils.read(self.OUTPUT_DIR)
+
+    def run(self):
+        # Parse generic data
+        possible_prefixes = ['m_str', 'm_map', 'm_n', 'm_fl', 'm_', 'fl', 'E', 'n']
+        parsed_generics = remove_prefixes(self.generic_data, possible_prefixes)
+
+        # Read existing generic data
+        existing_generics = self._read()
+
+        # If there ie existing data, validate the structure
+        if existing_generics is not None:
+            structure_keys_to_validate = ['ObjectiveParams', 'RejuvParams', 'ItemPricePerTier']
+            invalid_keys = validate_structures(
+                existing_generics, parsed_generics, structure_keys_to_validate
+            )
+            if len(invalid_keys) > 0:
+                print(
+                    '*WARNING* A structure within Generic data:'
+                    + f' {invalid_keys} is now different. Please verify the changes,'
+                    + ' and update the frontend page [[Module:GenericData]] accordingly.'
+                )
+
```

Comment:
I removed half of `generic-data.json` and it didn't throw a single warning.

Also with how the release pipeline will work automatically, I doubt these messages will even be seen

Reply:
Good catch, should be fixed now; before it was only checking if all the keys in 1 dict existed in the other. Now it should be checking in both directions. Didn't think to test that part earlier for some reason, was only testing changing the name of keys instead of outright removing them, and it not checking in both directions meant I was accidentally picking the flawless direction each time lol.

Also recall that for the 1st-layer of the dict, its only looking at keys in `structure_keys_to_validate`

Didn't think about the release pipeline. We need to ensure it handles errors well, as we should expect them to occur often when Valve changes structures. What idea do you propose for this? The first idea that pops into my head is a `output-data/warnings.txt/` file. `deadbot.py` would remove this file on every run to ensure it only exists if there's warnings from the previous runtime.